models/Account.php

```diff
@@ -242,4 +234,44 @@ public function getInvestor()
     {
         return $this->hasOne(User::class, ['id' => 'investor_id']);
     }
+
+    public function revertTransactions()
+    {
+        // revert all transactions
+        foreach (
+            Transaction::find()
+                ->where(['to_account_id' => $this->id])
+                ->orWhere(['from_account_id' => $this->id])->all()
+            as $transaction
+        ) {
+            $revertTransaction = new Transaction();
+
+            $revertTransaction->asset_id = $transaction->asset_id;
+            $revertTransaction->transaction_type = $transaction->transaction_type;
+            $revertTransaction->amount = $transaction->amount;
+            $revertTransaction->comment = $transaction->comment ? "$transaction->comment - Transaction Reverted" : "Campaign issue transaction reverted";
+            $revertTransaction->from_account_id = $transaction->to_account_id;
+
+            if($transaction->transaction_type == Transaction::TRANSACTION_TYPE_ISSUE){
+                // send coin to default account rather than issue account
+                $defaultAccount = Account::findOne([
+                    'space_id' => $this->getSpace()->one()->id,
+                    'account_type' => Account::TYPE_DEFAULT
+                ]);
+
+                $revertTransaction->to_account_id = $defaultAccount->id;
+
+            } else {
+                $revertTransaction->to_account_id = $transaction->from_account_id;
+            }
+
+            $revertTransaction->save();
+        }
+    }
+
+    public function disable()
+    {
+        $this->revertTransactions();
+        // TODO disable account
```

Comment:
remove TODO

Reply:
I need to keep , cause i need to implement it later when implementing deactivate account feature